src/app/components/kudos-give-kudos-small/kudos-give-kudos-small.component.js

```diff
@@ -14,11 +15,18 @@
             $scope.id = this.index;
         };
 
-        $scope.selectAutoText = function (text) {
-            $scope.sendKudosTo = text;
-            $scope.searchTermSelected = true;
-            $scope.autocompleteHide = true;
-        };
+        $scope.$watch(function () {
```

Comment:
There seem to be very many `$scope.$watch`es around the place - is there no way to use more of a **tell, don't ask** pattern, where you don't rely on random variable changes so much? I know it would require changing some of the architecture, but `$scope.$watch` should be used as a last resort, since it isn't the most performant solution and makes the testing really awkward.